app/javascript/controllers/media_player_controller.js

```diff
@@ -20,27 +20,61 @@ export default class extends Controller {
         const event = new CustomEvent('time-update', { detail: timestamp })
         window.dispatchEvent(event)
       })
-     
-      // When at least one of the tracks has been loaded, trigger transcript loading
-      // Before the tracks are loaded, the cues will not be recognized and the transcript will
-      // not load properly
-      this.player.textTracks().on('addtrack', () => {
-        // Retrieve the track objects.  Unfortunately, the "textTracks()" method did not cooperate
-        const tracks = this.player.textTracks_?.tracks_
-        // If there is at least one track we can attach the "loadeddata" event to
-        if(tracks && tracks.length > 0) {
-          this.player.textTracks_.tracks_[0].on('loadeddata', () => {
-            // Trigger this event, which will then lead to the transcript player load method
-            const event = new CustomEvent('media-data-loaded', { detail: this.player })
-            window.dispatchEvent(event)
-          })
-        }
-      })
+    })
+
+    // The loadeddata event occurs when the first frame of the video is available, and 
+    // happens after loadedmetadata
+    this.player.on('loadeddata', () => {
+      //At this point, we should have some track information, even if the cues have not been loaded
+      if(this.hasCaptionTracks()) {
+        const cuesAvailable = this.hasCaptionTrackCues(this.firstCaptionTrack())
+        if(cuesAvailable) {
+          // Trigger the media-data-loaded event which will then lead to the transcript load method
+          const event = new CustomEvent('media-data-loaded', { detail: this.player })
+          window.dispatchEvent(event)
+        } else {
+          // If the cues are not available, try iterating every half second a maximum of four times
+          let intervalTries = 0;
+          const intervalId = window.setInterval(() => {
+            intervalTries += 1
+            if(_this.hasCaptionTrackCues(this.firstCaptionTrack())) {
+              const event = new CustomEvent('media-data-loaded', { detail: _this.player })
```

Comment:
here too.

Reply:
Same as above - code has been updated, so I'll leave the first comment for now.